Algorithms.Tests/Problems/CoinChange/Dynamic/GenerateSingleCoinChangesTests.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Linq;
+using Algorithms.Problems.CoinChange;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Algorithms.Tests.Problems.CoinChange.Dynamic
+{
+    [TestFixture]
+    public class GenerateSingleCoinChangesTests
+    {
+        [Test]
+        public void GenerateSingleCoinChangesTests_Success()
+        {
+            DynamicCoinChangeSolver
+                .GenerateSingleCoinChanges(6, new[] { 1, 2, 3 })
+                .SequenceEqual(new[] { 3, 4, 5 })
+                .Should().BeTrue();
+
+            DynamicCoinChangeSolver
+                .GenerateSingleCoinChanges(10, new[] { 1, 2, 3, 7, 12, 15, 14 })
+                .SequenceEqual(new[] { 3, 7, 8, 9 })
+                .Should().BeTrue();
+
+            DynamicCoinChangeSolver
+                .GenerateSingleCoinChanges(1, new[] { 1, 2, 3, 7, 12, 15, 14 })
+                .SequenceEqual(new[] { 0 })
+                .Should().BeTrue();
+
+            DynamicCoinChangeSolver
+                .GenerateSingleCoinChanges(2, new[] { 1, 2, 3, 7, 12, 15, 14 })
+                .SequenceEqual(new[] { 0, 1 })
+                .Should().BeTrue();
+        }
+
+        [Test]
+        public void GenerateSingleCoinChangesTests_ShouldThrow_CoinCannotBeLesserOrEqualZero()
+        {
+            const int coin = 0;
+            var arr = new[] { 1, 2, 3 };
+
+            Func<int[]> act = () => DynamicCoinChangeSolver.GenerateSingleCoinChanges(coin, arr);
+
+            act.Should().Throw<InvalidOperationException>()
+                .WithMessage($"Coin cannot be lesser or equal to zero {nameof(coin)}.");
+        }
+
+        [Test]
+        public void GenerateSingleCoinChangesTests_ShouldThrow_CoinsArrayCannotBeNull()
+        {
+            const int coin = 10;
+            int[] coinsAsArray = default!;
+
+            Func<int[]> act = () => DynamicCoinChangeSolver.GenerateSingleCoinChanges(coin, coinsAsArray);
+
+            act.Should().Throw<ArgumentNullException>()
+                .WithMessage("Value cannot be null. (Parameter 'enumerableCoins')");
+        }
```

Comment:
No need in this test and the null check, the method signature clearly says that nulls are not allowed